keylime/tpm/ec_crypto_helper.py

```diff
@@ -0,0 +1,103 @@
+# pylint: disable=protected-access
+
+from typing import Optional
+
+from cryptography.hazmat import backends
+from cryptography.hazmat.backends.openssl.backend import Backend as OpenSSLBackend
+from cryptography.hazmat.primitives.asymmetric.ec import EllipticCurvePrivateKey, EllipticCurvePublicKey
+
+
+class EcCryptoHelper:
+    instance: Optional["EcCryptoHelper"] = None
+
+    @staticmethod
+    def get_instance() -> "EcCryptoHelper":
+        """Create and return a singleton EcCryptoHelper"""
+        if not EcCryptoHelper.instance:
+            EcCryptoHelper.instance = EcCryptoHelper()
+        return EcCryptoHelper.instance
+
+    def __init__(self) -> None:
+        self.backend = backends.default_backend()
+        if not isinstance(self.backend, OpenSSLBackend):
+            raise Exception("The python cryptography default backend must be using OpenSSL")
+
+        # sanity check the backend
+        if "_lib" not in dir(self.backend):
```

Comment:
Also this should be done with `hasattr()`

Reply:
I cannot respond to your comment above...

> Not sure to understand why a single instance is needed. Is because the init checks are too costly?

There's no need to check the interface more than once.

Reply:
> I cannot respond to your comment above...

You mean the `hasattr()`? If so is that `dir()` is not really expected to be used in code but in REPL. The natural code would be something like `if not hasattr(self.backend, "_lib")`

Reply:
Switch to hasattr() now. Would squash the top 3 patches at some point.